dogstatsd.py

```diff
@@ -542,6 +620,45 @@ def init(config_path=None, use_watchdog=False, use_forwarder=False, args=None):
     return reporter, server, c
 
 
+def init6(config_path=None, args=None):
+    c = get_config(parse_args=False, cfg_path=config_path)
+
+    if (not c['use_dogstatsd'] and
+            (args and args[0] in ['start', 'restart'] or not args)):
+        log.info("Dogstatsd is disabled. Exiting")
+        # We're exiting purposefully, so exit with zero (supervisor's expected
+        # code). HACK: Sleep a little bit so supervisor thinks we've started cleanly
+        # and thus can exit cleanly.
+        sleep(4)
+        sys.exit(0)
+
+    if not _is_affirmative(c.get('dogstatsd6_enable', False)):
+        log.debug("Dogstatsd v6 is disabled")
+        return None, None, None
+
+
+    env = copy.deepcopy(os.environ)
```

Comment:
If running alongside agent5, shouldn't we specify `DD_ENABLE_METADATA_COLLECTION = false` to tell dsd6 not to send host metadata?